firmware/ci.c

```diff
@@ -404,23 +419,161 @@ static void video_mode_list(void)
 
 static void video_mode_set(int mode)
 {
-	char mode_descriptors[PROCESSOR_MODE_COUNT*PROCESSOR_MODE_DESCLEN];
+	char mode_descriptor[PROCESSOR_MODE_DESCLEN];
```

Comment:
This is now a single value because I am only generating a single mode string, rather than generating all of the modes and then using one.